app/main.py

```diff
@@ -1,3 +1,128 @@
-def shop_trip():
-    # write your code here
-    pass
+import datetime
+import json
+import os.path
+
+from app.customer import Customer
+from app.shop import Shop
+from app.car import Car
+
+
+def calculate_gasoline_cost(
+    customer: Customer, shop: Shop, fuel_price: float
+) -> float:
+    km = (
+        round(
+            (
+                (customer.location[0] - shop.location[0]) ** 2
+                + (customer.location[1] - shop.location[1]) ** 2
+            )
+            ** 0.5,
+            2,
+        )
+        * 2
+    )
+    price = (customer.car.fuel_consumption / 100 * km) * fuel_price
+    return price
+
+
+def products_price(customer: Customer, shop: Shop) -> float:
+    return sum(
+        customer.product_cart[product] * shop.products[product]
+        for product in customer.product_cart.keys()
+    )
+
+
+def print_purchased_items_and_their_cost(
+        customer: Customer, shop: Shop
+) -> None:
+    for product in customer.product_cart:
+        print(
+            f"{customer.product_cart[product]} {product}s for "
+            f"{customer.product_cart[product] * shop.products[product]} "
+            f"dollars"
+        )
+
+
+def shop_trip() -> None:
+    path = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "config.json"
+    )
+
+    with open(path, "r") as json_data:
+        data = json.load(json_data)
+
+        fuel_price = data["FUEL_PRICE"]
+        customers_list = [
+            Customer(
+                name=customer["name"],
+                product_cart=customer["product_cart"],
+                location=customer["location"],
+                money=customer["money"],
+                car=Car(
+                    brand=customer["car"]["brand"],
+                    fuel_consumption=customer["car"]["fuel_consumption"],
+                ),
+            )
+            for customer in data["customers"]
+        ]
+        shop_list = [
+            Shop(
+                name=shop["name"],
+                location=shop["location"],
+                products=shop["products"],
+            )
+            for shop in data["shops"]
+        ]
+        prices_dict = {}
+
+    for customer in customers_list:
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in shop_list:
+            trip_price = round(round(calculate_gasoline_cost(
+                customer, shop, fuel_price
+            ) + products_price(customer, shop), 3), 2)
```

Comment:
There is an unnecessary nested `round()` function. Consider removing the inner `round()`.

Reply:
i need both functions to round the numbers 28.215472 and 17.145232 to 28.21 and 17.14 if i use at once round(number, 2), i get 28.22 and 17.15.
math.floor() and math.ceil() also do not suit me, since other numbers should be rounded according to mathematical rules, so I found such a solution
the previous problem  solution was even worse:
            if str(round(trip_price, 3))[-1] == "5":
                trip_price = float(str(round(trip_price, 3))[:-1])
            else:
                trip_price = round(trip_price, 2)